app/user.go

```diff
@@ -2495,7 +2495,28 @@ func (a *App) UpdateThreadReadForUserByPost(currentSessionId, userID, teamID, th
 		return nil, model.NewAppError("UpdateThreadReadForUser", "app.user.update_thread_read_for_user_by_post.app_error", nil, "", http.StatusBadRequest)
 	}
 
-	return a.UpdateThreadReadForUser(currentSessionId, userID, teamID, threadID, post.CreateAt-1)
+	lastViewed := post.CreateAt - 1
+	if post.UserId == userID {
+		posts, nErr := a.Srv().Store.Thread().GetPosts(threadID, post.CreateAt-1)
```

Comment:
I think this should be

```suggestion
		posts, nErr := a.Srv().Store.Thread().GetPosts(threadID, post.CreateAt)
```

since we fetch posts greater than or equal from the timestamp https://github.com/mattermost/mattermost-server/blob/master/store/sqlstore/thread_store.go#L914

Reply:
I think that we'll need to sort posts by `CreateAt`, it seems that they are not sorted.